src/Module/Admin/Site.php

```diff
@@ -332,8 +334,8 @@ protected function content(array $request = []): string
 
 		if (DI::config()->get('system', 'directory_submit_url') &&
 			!DI::config()->get('system', 'directory')) {
-			DI::config()->set('system', 'directory', dirname(DI::config()->get('system', 'directory_submit_url')));
-			DI::config()->delete('system', 'directory_submit_url');
+			DI::config()->set('system', 'directory', dirname(DI::config()->get('system', 'directory_submit_url')), false);
+			DI::config()->delete('system', 'directory_submit_url', false);
```

Comment:
These are never saved.
```suggestion
			DI::config()->set('system', 'directory', dirname(DI::config()->get('system', 'directory_submit_url')), false);
			DI::config()->delete('system', 'directory_submit_url', false);
			DI::config()->save();
```